File: Modules/TaskFlow/TaskState/TaskStateColumns.cs
namespace TaskFlowSerenity.TaskFlow.Columns;

[ColumnsScript("TaskFlow.TaskState")]
[BasedOnRow(typeof(TaskStateRow), CheckNames = true)]
public class TaskStateColumns
{
    [EditLink]
    public string Name { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Modules/ServerTypes/TaskFlow/TaskStateColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TaskStateRow } from "./TaskStateRow";

export interface TaskStateColumns {
    Name: Column<TaskStateRow>;
    CreatedAt: Column<TaskStateRow>;
    UpdatedAt: Column<TaskStateRow>;
}

export class TaskStateColumns extends ColumnsBase<TaskStateRow> {
    static readonly columnsKey = 'TaskFlow.TaskState';
    static readonly Fields = fieldsProxy<TaskStateColumns>();
}